lib/open3.rb

```diff
@@ -899,48 +899,75 @@ def capture2e(*cmd)
   end
   module_function :capture2e
 
-  # Open3.pipeline_rw starts a list of commands as a pipeline with pipes
-  # which connect to stdin of the first command and stdout of the last command.
-  #
-  #   Open3.pipeline_rw(cmd1, cmd2, ... [, opts]) {|first_stdin, last_stdout, wait_threads|
-  #     ...
-  #   }
+  # :call-seq:
+  #   Open3.pipeline_rw([env, ] *cmds, options = {}) -> [first_stdin, last_stdout, wait_threads]
   #
-  #   first_stdin, last_stdout, wait_threads = Open3.pipeline_rw(cmd1, cmd2, ... [, opts])
-  #   ...
-  #   first_stdin.close
-  #   last_stdout.close
+  # Basically a wrapper for
+  # {Process.spawn}[https://docs.ruby-lang.org/en/master/Process.html#method-c-spawn]
+  # that:
   #
-  # Each cmd is a string or an array.
-  # If it is an array, the elements are passed to Process.spawn.
+  # - Creates a child process for each of the given +cmds+
+  #   by calling Process.spawn.
+  # - Pipes the +stdout+ from each child to the +stdin+ of the next child,
+  #   or, for the first child, from the caller's +stdin+,
+  #   or, for the last child, to the caller's +stdout+.
+  # - Waits for all child processes to exit.
```

Comment:
I missed it in the past reviews, but I don't think any of the pipeline methods wait for the child processes to exit. For example:

```ruby
require "open3"

stdin, stdout, thr = Open3.pipeline_rw("ruby")

stdin.write("puts 'hello world'")
stdin.close
puts stdout.read
```

Here, `Open3#pipeline_rw` doesn't wait for the Ruby process to exit because I feed it the program on the next line. You can see it executes because it outputs "Hello world".

Reply:
Fixed (except that (I think) pipeline does wait).